### PR TITLE
use more efficient hashes when talking to kubernetes

### PR DESCRIPTION
**Pull request by grosser**
more details here https://github.com/abonas/kubeclient/issues/251

### CODE REVIEW

**Comment by jcantrill** [comment]
@grosser started through your PRs from oldest... looks like we need a rebase here

**Comment by grosser** [comment]
rebased!

File: lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb
Comment (cben) [inline lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb:100]: :+1: this a special case where the change is not 1-to-1:

> With default (as: :ros) return format, the returned object acts like an array of the individual pods, but also supports a .resourceVersion method.

> With :parsed and :parsed_symbolized formats, the returned data structure matches kubernetes list structure: it's a hash containing metadata and items keys, the latter containing the individual pods.

File: test/plugin/test_watch_pods.rb
Comment (cben) [inline test/plugin/test_watch_pods.rb:29]: What do you mean? In test?  In real life?

File: test/plugin/watch_test.rb
Comment (cben) [inline test/plugin/watch_test.rb:53]: This one is a bit strange.  It's used in 2 tests like this:
```
        @client.stub :watch_namespaces, [[@created, @exception_raised]] do
```
Is the array used to simulate the watch object .each first yielding a value, then raising an exception? But instead of the watch object raising, it yields an "explosive" value that raises when it gets accessed?

**Comment by jcantrill** [comment]
I'll defer to responding to @cben before merging.

File: test/plugin/test_watch_pods.rb
Comment (grosser) [reply test/plugin/test_watch_pods.rb:29]: fixed, was a leftover comment from debugging

File: test/plugin/watch_test.rb
Comment (grosser) [reply test/plugin/watch_test.rb]: made this a bit more obvious ... it was meant to cause an exception when being used, since "correct" notices do not blow up, but any undefined behavior is rescued

**Comment by grosser** [comment]
@cben looks good ?

**Review by cben — APPROVED** [review]
LGTM :100:

**Comment by jcantrill** [comment]
@grosser looks like there is a test failure. please investigate

**Comment by grosser** [comment]
rebuild fixed it ... some random error :)

**Comment by grosser** [comment]
@jcantrill 👋

**Pull request merged**
